hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/JetConventions.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.opt;
+
+import com.hazelcast.jet.sql.impl.opt.logical.LogicalRel;
+import com.hazelcast.jet.sql.impl.opt.physical.PhysicalRel;
+import org.apache.calcite.plan.Convention;
+import org.apache.calcite.plan.RelTraitSet;
+
+public final class JetConventions {
+
+    public static final Convention LOGICAL = new Convention.Impl("LOGICAL", LogicalRel.class);
+
+    public static final Convention PHYSICAL = new Convention.Impl("PHYSICAL", PhysicalRel.class) {
+        @Override
+        public boolean canConvertConvention(Convention toConvention) {
+            return true;
```

Comment:
This code is copy-paste from the IMDG planner. Please beware that `PHYICAL.canConvertConvention=true` is a kind of hack to simulate bottom-up optimization. We borrowed this idea from the Apache Drill project. Without it, the optimizer cannot derive the physical properties of children nodes and hence cannot plan optimal physical implementations. The side effect of this is excessive invocations of physical rules, which could become a problem when we introduce joins.

I analyzed other possible solutions to the problem recently. Another approach is to use a custom converter rule (instead of the new `AbstractConverter.ExpandConversionRule` that is used in `PhysicalRules`) the pulls physical properties in a top-down way. Apache Flink works this way. In the future, we will need to understand which of this approach is better for complex queries. So there is a risk that we will have to rework it from "Drill-way" to "Flink-way". 

I do not propose any changes now, just be aware of this possibility.

Reply:
Hmm, `PHYICAL.canConvertConvention=true` does not seem to have any influence on Jet, setting it to `false` does not break any tests...

Reply:
Maybe the test queries are not complicated enough, or maybe something changed in Apache Calcite. In IMDG the problem manifested in complex TPC-H queries. I tried to change it to `false`, and tests also worked. But we do not have join planning yet, this is where the problem may re-appear. I think just demonstrates once again that we should better understand what we are doing here in the future :-)